node/src/test_utils/recorder.rs

```diff
@@ -952,8 +964,31 @@ mod tests {
     }
 
     #[test]
-    fn counter_msgs_of_same_type_are_checked_sequentially_and_triggered_by_first_matching_id_method(
-    ) {
+    fn counter_msgs_evaluate_lazily_so_the_msgs_with_the_same_triggers_are_eliminated_sequentially()
+    {
+        // This test demonstrates the need for caution in setups where multiple messages are sent
+        // at different times and should be responded to by different counter-messages. However,
+        // the trigger methods of these setups also apply to each other. Which setup gets
+        // triggered depends purely on the order used to supply them to the recorder
+        // in SetUpCounterMsgs.
```

Comment:
I was wondering, if it should include the information somewhere where the utility is declared that,

"In case the trigger methods are applicable to multiple counter messages, the triggered message will depend on the order the messages were supplied in the test using SetUpCounterMsgs. To avoid any confusions regarding reordering, please use `MsgIdentification::ByMatch` or `MsgIdentification::ByPredicate` instead of `MsgIdentification::ByTypeId`.